subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecSpec.java

```diff
@@ -73,68 +66,51 @@ static void copyBaseExecSpecTo(BaseExecSpec source, BaseExecSpec target) {
     }
 
     @Override
-    public List<String> getCommandLine() {
-        return argumentsSpec.getCommandLine();
+    public Provider<List<String>> getCommandLine() {
+        return getExecutable().zip(getArgs(), (SerializableLambdas.SerializableBiFunction<String, List<String>, List<String>>) (executable, args) -> {
```

Comment:
❓ Why do we need this mess with the serializable lambdas? Is this something regular users need to do when using `zip()`, too?

Reply:
If we don't add it, this fails at configuration cache. Users don't need to do that, since we have instrument their lambdas with [LambdaSerializationTransformer](https://github.com/gradle/gradle/blob/d8f7224f3144a929b986cf8f9b81523489923b64/subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/LambdaSerializationTransformer.java#L59).

Reply:
That explains, thanks. I wonder how we could automate this. It seems pretty easy to make the mistake of not adding the serializable-lambda stuff...